### ISSUE TITLE
Spaces Bucket not found on apply ACL

### ISSUE THREAD

**Issue by vinicius73** [digitalocean/terraform-provider-digitalocean#510]
### Terraform Version
```
Terraform v0.13.5
+ provider registry.terraform.io/digitalocean/digitalocean v2.0.1
```

### Affected Resource(s)
- digitalocean_spaces_bucket

### Terraform Configuration Files

```hcl
resource "digitalocean_spaces_bucket" "index-key" {
  name   = "x-space-name-x"
  region = "nyc3"
  acl    = "private"
}
```

### Debug Output

```
2020-10-26T18:13:47.380-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1: 2020/10/26 18:13:47 Bucket created
2020-10-26T18:13:47.380-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1: 2020/10/26 18:13:47 [DEBUG] Spaces put bucket ACL: {
2020-10-26T18:13:47.380-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1:   ACL: "private",
2020-10-26T18:13:47.381-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1:   Bucket: "x-space-name-x"
2020-10-26T18:13:47.381-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1: }
2020-10-26T18:13:47.381-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1: 2020/10/26 18:13:47 [DEBUG] Waiting for state to become: [success]
2020-10-26T18:13:47.535-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1: 2020/10/26 18:13:47 [DEBUG] Waiting for state to become: [success]
2020-10-26T18:13:47.656-0300 [DEBUG] plugin.terraform-provider-digitalocean_v2.0.1: 2020/10/26 18:13:47 [WARN] Spaces Bucket (x-space-name-x) not found, error code (404)
2020/10/26 18:13:47 [DEBUG] module.production-cluster.module.storage.digitalocean_spaces_bucket.this["index-key"]: apply errored, but we're indicating that via the Error pointer rather than returning it: Provider produced inconsistent result after apply: When applying changes to module.production-cluster.module.storage.digitalocean_spaces_bucket.this["index-key"], provider "registry.terraform.io/digitalocean/digitalocean" produced an unexpected new value: Root resource was present, but now absent.
```

### Expected Behavior
Normal creation of resource and state

### Actual Behavior
The resource was created, but state is inconsistent.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by andrewsomething** [comment]
@vinicius73, thanks for flagging this for us. In my testing, it looks like this is only effecting `nyc3`. The example config is working as expected for me in other regions. Can you (or anyone else effected) confirm if that is the case for you as well?

**Comment by andrewsomething** [comment]
The Spaces team identified the issue in NYC3. Additionally, we've added a change here that should protect against any future instability like this. The fix is part of the 2.0.2 release. You can update with:

    terraform init -upgrade

**Comment by vinicius73** [comment]
Thanks @andrewsomething
